src/extra/gpkg.jl

```diff
@@ -0,0 +1,179 @@
+  # with the srs_id specified for the column by the
+  # gpkg_geometry_columns table srs_id column value.
+  org = metadata.org
+  code = metadata.code
+  srsid = metadata.srsid
+  if srsid == 0
+    crs = LatLon{NoDatum}
+  elseif srsid == 4326
+    crs = LatLon{WGS84Latest}
+  elseif srsid == -1
+    crs = Cartesian{NoDatum}
+  else
+    if org == "EPSG"
+      crs = CoordRefSystems.get(EPSG{code})
+    elseif org == "ESRI"
+      crs = CoordRefSystems.get(ESRI{code})
+    end
+  end
+
+  # According to https://www.geopackage.org/spec/#r14
+  # The table_name column value in a gpkg_contents table row 
+  # SHALL contain the name of a SQLite table or view.
+  tablename = metadata.tablename
+
+  # According to https://www.geopackage.org/spec/#r30
+  # A feature table or view SHALL have only one geometry column.
+  geomcolumn = metadata.geomcolumn |> Symbol
+
+  # Retrieve names of columns with attributes (i.e., ≠ geometry)
+  # pk is the index of the column within the primary key,
+  # or 0 for columns that are not part of the primary key
+  tabinfo = SQLite.tableinfo(db, tablename)
+  columns = [Symbol(name) for (name, pk) in zip(tabinfo.name, tabinfo.pk) if pk == 0]
+  attribs = setdiff(columns, [geomcolumn])
+
+  # load feature table from database
+  gpkgtable = DBInterface.execute(db, "SELECT $(join(columns, ',')) FROM $tablename;")
+
+  # extract attribute table and geometries
+  pairs = map(Tables.rows(gpkgtable)) do row
+    # retrieve attribute values as a named tuple
+    vals = (; (col => Tables.getcolumn(row, col) for col in attribs)...)
+
+    # retrieve geometry binary data
+    geomcol = Tables.getcolumn(row, geomcolumn)
+
+    ismissing(geomcol) && return vals, missing
+
+    # retrieve geometry binary data as IO buffer or missing
+    buff = IOBuffer(geomcol)
+
+    # skip GPKG binary header bytes
+    skipgpkgheader!(buff)
+
+    # convert WKB geometry into Meshes.jl geometry
+    geom = wkb2meshes(buff, crs)
+
+    vals, geom
+  end
+
+  # filter missing geometries and handle tables without attributes
+  filter!(pair -> !ismissing(last(pair)), pairs)
+  table = isempty(attribs) ? nothing : first.(pairs)
+  geoms = GeometrySet(last.(pairs))
+
+  table, geoms
+end
+
+# According to https://www.geopackage.org/spec/#r19
+# A GeoPackage SHALL store feature table geometries in
+# SQL BLOBs using the Standard GeoPackageBinary format
+function skipgpkgheader!(buff)
+  # check the GeoPackageBinaryHeader for the 'GP' magic
+  read(buff, UInt16) == 0x5047 || @warn "Missing magic 'GP' string in GPkgBinaryGeometry"
+
+  # skip version (0 => version 1)
+  skip(buff, 1)
+
+  # bit layout of GeoPackageBinary flag byte
+  # https://www.geopackage.org/spec/#flags_layout
+  # ---------------------------------------
+  # bit # 7 # 6 # 5 # 4 # 3 # 2 # 1 # 0 #
+  # use # R # R # X # Y # E # E # E # B #
+  # ---------------------------------------
+  # R: reserved for future use; set to 0
+  # X: GeoPackageBinary type
+  # Y: empty geometry flag
```

Comment:
Is this Y bit related to the missing geometry case we handled in the other function?